src/InvestmentManager.sol

```diff
@@ -255,6 +244,9 @@ contract InvestmentManager is Auth {
         lpValues.maxWithdraw = lpValues.maxWithdraw + currencyPayout;
         lpValues.maxRedeem = lpValues.maxRedeem + trancheTokensPayout;
 
+        escrow.approve(_currency, address(userEscrow), currencyPayout);
```

Comment:
I think we can make that a general approval when we add a new supported currency